android-project/app/src/main/java/org/uzebox/cuzebox/app/CUzeBoxActivity.java

```diff
@@ -6,4 +6,29 @@
  * A sample wrapper class that just calls SDLActivity 
  */ 
 
-public class CUzeBoxActivity extends SDLActivity { }
+public class CUzeBoxActivity extends SDLActivity { 
+  
+ @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        if(Build.VERSION.SDK_INT>=23) {
+            //did user granted this ermission?
+            if (checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+                //nope, it's Chuck Testa
+
+                //should we show toast with info WHY do we ned this permission form user?
+                if(shouldShowRequestPermissionRationale(Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+                    final Toast info =  Toast.makeText(this, "Do it, albo urwę ci ryja", Toast.LENGTH_LONG);
```

Comment:
Could we put this text into the xml config, be easier to manage languages that way? :) Apart from that, this looks good.

Reply:
The permissions request is done asynchronously and the native code still ends up running in the background before the permissions are granted, so I ended up with a crash dialog obscuring the permissions request dialog.

I do have a proper fix for this, and when I'm free I'll commit it so that forces the native code to wait until permissions have been granted. Thanks for the contribution and feedback!